crates/oxc_linter/src/rules/jsdoc/check_tag_names.rs

```diff
@@ -0,0 +1,1047 @@
+              {
+                "typed": true,
+              },
+            ])),
+            None,
+        ),
+        (
+            "
+        			        const a = {
+        			          /** @abstract */
+        			          b: true,
+        			        };
+        			      ",
+            Some(serde_json::json!([
+              {
+                "typed": true,
+              },
+            ])),
+            None,
+        ),
+        (
+            "
+        			        /** @template */
+        			        let a;
+        			      ",
+            Some(serde_json::json!([
+              {
+                "typed": true,
+              },
+            ])),
+            None,
+        ),
+        (
+            "
+        			        /**
+        			         * Prior description.
+        			         *
+        			         * @template
+        			         */
+        			        let a;
+        			      ",
+            Some(serde_json::json!([
+              {
+                "typed": true,
+              },
+            ])),
+            None,
+        ),
+    ];
+
+    let dts_pass = vec![
+        (
+            "
+        			        /** @default 0 */
+        			        declare let a;
+        			      ",
+            Some(serde_json::json!([
+              {
+                "typed": true,
+              },
+            ])),
+            None,
+        ),
+        (
+            "
+        			        /** @abstract */
+        			        let a;
+        			      ",
+            Some(serde_json::json!([
+              {
+                "typed": true,
+              },
+            ])),
+            None,
+        ),
+        (
+            "
+        			        /** @abstract */
+        			        declare let a;
+        			      ",
+            Some(serde_json::json!([
+              {
+                "typed": true,
+              },
+            ])),
+            None,
+        ),
+        (
+            "
+        			        /** @abstract */
+        			        { declare let a; }
+        			      ",
+            Some(serde_json::json!([
+              {
+                "typed": true,
+              },
+            ])),
+            None,
+        ),
+        (
+            "
+        			        function test() {
+        			          /** @abstract */
+        			          declare let a;
+        			        }
+        			      ",
+            Some(serde_json::json!([
+              {
+                "typed": true,
+              },
+            ])),
+            None,
+        ),
+    ];
+    let dts_fail = vec![(
+        "
+        			        /** @typoo {string} (fail: invalid name) */
+        			        let a;
+        			      ",
+        None,
+        None,
+    )];
+
+    Tester::new(CheckTagNames::NAME, pass, fail).test_and_snapshot();
+    // Currently only 1 snapshot can be saved under a rule name
+    Tester::new(CheckTagNames::NAME, dts_pass, dts_fail).change_rule_path("test.d.ts").test();
```

Comment:
Do you have any ideas...?

Reply:
Snapshots are used for checking whether the spans are correct. `.test()` for other tests is fine.